mutation_order_gibbs.py

```diff
@@ -360,3 +425,60 @@ def _get_denom_update(self, old_denominator, prev_feat_idxs, feat_mut_step):
             prev_exp_theta_sum = self.exp_theta_sum[prev_feat_idxs] if len(prev_feat_idxs) else self.exp_theta_num_cols
             new_denom = old_denominator - prev_exp_theta_sum - np.sum(old_feat_exp_theta_sums) + np.sum(new_feat_exp_theta_sums)
         return float(new_denom)
+
+    def _get_risk_update(self, base_risk_vec, prev_mut_pos, feat_mut_step, new_denom):
+        """
+        Similar to _get_denom_update but calculate the updated risk vector to use as residuals
+
+        @param base_risk_vec: the vector of summands from denominator from the previous mutation step weighted with the new denominator
+        @param prev_mut_pos: previously mutated position that won't contribute to the updated risk group
+        @param feat_mut_step: the features that differed for this next mutation step
+        @param new_denom: new denominator
+        """
+        for feat_pos, feat_idxs in feat_mut_step.neighbors_feat_new.iteritems():
```

Comment:
do you need to update the positions that aren't neighbors of mutating positions because those positions have a new denominator (the numerator is the same though)?

Reply:
That's taken care of here:

https://github.com/matsengrp/samm/blob/7b9fbe9f38b5d3a209016f152b47db742362df1e/mutation_order_gibbs.py#L392

I think it's probably confusing though---I'll toy around with changing it to make it more clear.